tests/test_updater.py

```diff
@@ -0,0 +1,253 @@
+    sure that this fields becomes equal to the new version, so that the user is not
+    erroneously prompted to a version they already have.
+    """
+    # Store the settings of Dangerzone 0.4.2 to the filesystem.
+    old_settings = settings.Settings.generate_default_settings()
+    old_settings["updater"]["latest_version"] == "0.4.2"
+    save_settings(tmp_path, old_settings)
+
+    # Mimic an upgrade to version 0.4.3, by making Dangerzone report that the current
+    # version is 0.4.3.
+    expected_settings = default_updater_settings()
+    expected_settings["latest_version"] = "0.4.3"
+    monkeypatch.setattr(
+        settings, "get_version", lambda: expected_settings["latest_version"]
+    )
+
+    # Ensure that the Settings class will correct the latest version field to 0.4.3.
+    updater = generate_isolated_updater(tmp_path, monkeypatch)
+    assert updater.dangerzone.settings.get("updater") == expected_settings
+
+    # Simulate an updater check that found a newer Dangerzone version (e.g., 0.4.4).
+    expected_settings["latest_version"] = "0.4.4"
+    updater.dangerzone.settings.updater_set(
+        "latest_version", expected_settings["latest_version"]
+    )
+    updater.dangerzone.settings.save()
+
+    # Ensure that the Settings class will leave the "latest_version" field intact the
+    # next time we reload the settings.
+    updater.dangerzone.settings.load()
+    assert updater.dangerzone.settings.get("updater") == expected_settings
+
+
+# FIXME: This test should not fail
+@pytest.mark.xfail(reason="We haven't enabled this behavior in Linux yet")
+def test_linux_first_run(updater: Updater) -> None:
+    """Check that Linux users are not prompted for update checks."""
+    expected_settings = default_updater_settings()
+    expected_settings["first_run"] = False
+    expected_settings["check"] = False
+    assert updater.should_check_for_updates() == False
+    assert updater.dangerzone.settings.get("updater") == expected_settings
+
+
+# FIXME: Make this check run only on Windows/MacOS platforms.
+def test_user_prompts(
+    updater: Updater, monkeypatch: MonkeyPatch, mocker: MockerFixture
+) -> None:
+    """Test prompting users to ask them if they want to enable update checks."""
+    # First run
+    #
+    # When Dangerzone runs for the first time, users should not be asked to enable
+    # updates.
+    expected_settings = default_updater_settings()
+    expected_settings["first_run"] = False
+    expected_settings["check"] = None
+    assert updater.should_check_for_updates() == False
+    assert updater.dangerzone.settings.get("updater") == expected_settings
+
+    # Second run
+    #
+    # When Dangerzone runs for a second time, users can be prompted to enable update
+    # checks. Depending on their answer, we should either enable or disable them.
+    alert_mock = mocker.MagicMock()
+    monkeypatch.setattr(updater_mod, "Alert", alert_mock)
```

Comment:
You mean patch this method, instead of `Alert`?

https://github.com/freedomofpress/dangerzone/blob/e6cd583946e26d09c829afd286b4b2ce3a8f9f48/dangerzone/gui/updater.py#L91-L103

That's doable, but I don't want to be concerned about what will happen once we have more logic in this method. I don't have a strong preference though, and probably tests will fail in that case, so I can proceed with your suggestion, if you want to.

Reply:
I don't know where I wanted to go with that. My original comment makes no sense. Sorry. Just ignore it. Probably it was some confusion with patching the module (which you explained your reasoning for somewhere else).